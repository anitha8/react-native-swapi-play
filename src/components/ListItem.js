import PropTypes from "prop-types";
import React, { Component } from "react";
import {
    Text,
    View,
    TouchableOpacity, StyleSheet,
} from "react-native";

class ListItem extends Component {
    static propTypes = {
        selectedItemFlag: PropTypes.bool,
        onPress: PropTypes.func.isRequired,
        itemText: PropTypes.string.isRequired,
        gradientColor: PropTypes.string
    };

    static defaultProps = {
        selectedItemFlag: false,
        onPress: () => { },
        itemText: "",
        gradientColor: '#FF0000',
    };

    render() {
        const {
            selectedItemFlag,
            onPress,
            itemText,
            gradientColor
        } = this.props;
        return (
            <View>
                <TouchableOpacity
                    onPress={onPress}>
                    <View style={[styles.rowItems, {backgroundColor: gradientColor}]}>
                        <Text
                            style={[
                                styles.disabledMenu,
                                selectedItemFlag ? { color: 'grey' } : { color: 'black' },
                            ]}>
                            {itemText}
                        </Text>
                    </View>
                </TouchableOpacity>
            </View>
        );
    }
}

const styles = StyleSheet.create({
    disabledMenu: {
        flex: 1,
        fontSize: 16,
        paddingTop: 5,
        paddingBottom: 5,
    },
    rowItems: {
        flexDirection: "row",
        alignItems: "center",
        height: 40,
        justifyContent:"center"
    },
});

export default ListItem;
