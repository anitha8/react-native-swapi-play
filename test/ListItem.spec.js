import { Text, TouchableOpacity } from "react-native";
import React from "react";
import { configure, shallow } from "enzyme";

import Adapter from "enzyme-adapter-react-16";
import ListItem from "../src/components/ListItem";

configure({ adapter: new Adapter() });

describe("Testing SelectDropDownList Component", () => {
    let wrapper;
    const props = {
        selectedItemFlag: false,
        onPress: jest.fn(),
        itemText: "Abuja",
    };

    beforeEach(() => {
        wrapper = shallow(<ListItem {...props} />);
    });

    it("Check ListItem Snapshot", () => {
        expect(wrapper.dive()).toMatchSnapshot();
    });

    it("should call onPress on item select", () => {
        const render = wrapper.dive();
        render.find(TouchableOpacity).simulate("press");
        expect(props.onPress).toBeCalled();
    });

    it("check itemText same as props", () => {
        const render = wrapper.dive();
        const text = render.find(Text).first().props().children;
        expect(text).toBe(props.itemText);
    });

    it("check selectedItemFlag toggle same as props", () => {
        const render = wrapper.dive();
        const text = render.find(TouchableOpacity);
        wrapper.setProps({ selectedItemFlag: true });
        const render1 = wrapper.dive();
        const text1 = render1.find(TouchableOpacity);
    });
});
