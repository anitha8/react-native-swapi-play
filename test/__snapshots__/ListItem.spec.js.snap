// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing SelectDropDownList Component Check ListItem Snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <View>
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[MockFunction]}
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 40,
              "justifyContent": "center",
            },
            Object {
              "backgroundColor": "#FF0000",
            },
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "flex": 1,
                "fontSize": 16,
                "paddingBottom": 5,
                "paddingTop": 5,
              },
              Object {
                "color": "black",
              },
            ]
          }
        >
          Abuja
        </Text>
      </View>
    </TouchableOpacity>
  </View>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <TouchableOpacity
        activeOpacity={0.2}
        onPress={[MockFunction]}
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "height": 40,
                "justifyContent": "center",
              },
              Object {
                "backgroundColor": "#FF0000",
              },
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "flex": 1,
                  "fontSize": 16,
                  "paddingBottom": 5,
                  "paddingTop": 5,
                },
                Object {
                  "color": "black",
                },
              ]
            }
          >
            Abuja
          </Text>
        </View>
      </TouchableOpacity>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "activeOpacity": 0.2,
        "children": <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "height": 40,
                "justifyContent": "center",
              },
              Object {
                "backgroundColor": "#FF0000",
              },
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "flex": 1,
                  "fontSize": 16,
                  "paddingBottom": 5,
                  "paddingTop": 5,
                },
                Object {
                  "color": "black",
                },
              ]
            }
          >
            Abuja
          </Text>
        </View>,
        "onPress": [MockFunction],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "flex": 1,
                  "fontSize": 16,
                  "paddingBottom": 5,
                  "paddingTop": 5,
                },
                Object {
                  "color": "black",
                },
              ]
            }
          >
            Abuja
          </Text>,
          "style": Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 40,
              "justifyContent": "center",
            },
            Object {
              "backgroundColor": "#FF0000",
            },
          ],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": "Abuja",
            "disabled": false,
            "ellipsizeMode": "tail",
            "style": Array [
              Object {
                "flex": 1,
                "fontSize": 16,
                "paddingBottom": 5,
                "paddingTop": 5,
              },
              Object {
                "color": "black",
              },
            ],
          },
          "ref": null,
          "rendered": "Abuja",
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
    "type": "View",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <TouchableOpacity
          activeOpacity={0.2}
          onPress={[MockFunction]}
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "height": 40,
                  "justifyContent": "center",
                },
                Object {
                  "backgroundColor": "#FF0000",
                },
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "flex": 1,
                    "fontSize": 16,
                    "paddingBottom": 5,
                    "paddingTop": 5,
                  },
                  Object {
                    "color": "black",
                  },
                ]
              }
            >
              Abuja
            </Text>
          </View>
        </TouchableOpacity>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "activeOpacity": 0.2,
          "children": <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "height": 40,
                  "justifyContent": "center",
                },
                Object {
                  "backgroundColor": "#FF0000",
                },
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "flex": 1,
                    "fontSize": 16,
                    "paddingBottom": 5,
                    "paddingTop": 5,
                  },
                  Object {
                    "color": "black",
                  },
                ]
              }
            >
              Abuja
            </Text>
          </View>,
          "onPress": [MockFunction],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "flex": 1,
                    "fontSize": 16,
                    "paddingBottom": 5,
                    "paddingTop": 5,
                  },
                  Object {
                    "color": "black",
                  },
                ]
              }
            >
              Abuja
            </Text>,
            "style": Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "height": 40,
                "justifyContent": "center",
              },
              Object {
                "backgroundColor": "#FF0000",
              },
            ],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": "Abuja",
              "disabled": false,
              "ellipsizeMode": "tail",
              "style": Array [
                Object {
                  "flex": 1,
                  "fontSize": 16,
                  "paddingBottom": 5,
                  "paddingTop": 5,
                },
                Object {
                  "color": "black",
                },
              ],
            },
            "ref": null,
            "rendered": "Abuja",
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
      "type": "View",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
